api/src/main/java/org/openmrs/obs/handler/TextHandler.java

```diff
@@ -84,7 +84,12 @@ public Obs getObs(Obs obs, String view) {
 		}
 		
 		Assert.notNull(complexData, "Complex data must not be null");
-		complexData.setMimeType("text/plain");
+		
+		// Get the Mime Type and set it
+		String mimeType = getMimeTypeFromFile(file);
+		mimeType = mimeType != null ? mimeType : "application/octet-stream";
```

Comment:
Of the two "application/octet-stream" and "text/plain", which is more appropriate for this class?

Reply:
You're right, I'll change this.